text/1180971/async_korach_moran_zaks.tex

```diff
@@ -0,0 +1,213 @@
+\end{proof}
+\begin{lemma}
+If some King is awake he will send an ASK message to a node not in his kingdom.
+\end{lemma}
+\begin{proof}
+The king is sending messages to all Rivals. If he sends message to his kingdom he will receive YOUR\_CITIZEN message which will cause the king to send another message.
+\end{proof}
+\begin{lemma}
+If \(king_s\) send a message to another kingdom of \(king_j\), \(king_j\) will become a citizen. 
+\end{lemma}
+\begin{proof}
+If \(king_j\) directly receives the message he immediately become the citizen. Otherwise it will be transferred by Vassals to the king. The message could be blocked on the way to \(king_j\) but that means that the node that blocked the message is waiting for UPDATE from his king as a response for ASK message with higher status than status(\(king_j\)). It means that one of the messages will cause the 
+\(king_j\) to become the citizen.
+\end{proof}
+By Lemma 3 we get contradiction to the assumptions that \(s > 1\)
+The algorithm will stop after the King of Kings announce his presence (by sending the I\_AM\_THE\_KING message down the tree).
+
+\section*{Proof of complexity}
+\begin{theorem}
+The number of messages sent is \(5n\log_2(n) + \mathcal{O}(n)\)
+\end{theorem}
+\begin{lemma}
+When the algorithm stops the biggest possible phase is bounded by \(\lfloor \log_2(n) \rfloor\)
+\end{lemma}
+\begin{proof}
+Whenever a king at phase \(t\) 
+increases his phase, he annexes another 
+king of phase \(t\). Therefore, we have at 
+most \(\frac{n}{2}\) kings in phase 1, \(\frac{n}{2^2}\) kings in  phase 2 ... \(\frac{n}{2^l}\) kings in phase \(l\), for every \(1 \leq l \leq \log_2(n)\)
+\end{proof}
+\begin{proof}{Theorem 2}
+Let's give upper bound for every type of messages
+\begin{enumerate}
+    \item I\_AM\_THE\_KING - exactly \(n-1\) messages
+    \item YOUR\_CITIZEN - each node sends at most one such message - as a reply to an ASK message - per phase. The total number of such messages is bounded by \(n\log_2(n)\)
+    \item ASK - a king can send at most len(King.Vassals) + 1, therefore all the kings in this phase sent together at most n messages, the total number of messages sent by kings is bounded by \(n\log_2(n)\), Every citizen transfers at most one ASK message per phase, the total number of such messages sent by all citizens is also bounded by \(n\log_2(n)\) 
+    \item ACCEPT - Citizens send no more than ASK messages - bounded by \(n\log_2(n)\), Kings send exactly \(n-1\) messages
+    \item UPDATE - one such message per phase - bounded by  \(n\log_2(n)\) 
+\end{enumerate}
+    So the total number of messages is bounded by \(5n\log_2(n) + 2n\). This bound is pretty loose. The algorithm uses around 800 messages for n = 100 but the bound is equal to 3200.
+\end{proof}
+
+\begin{thebibliography}{9}
+\bibitem{article}
+Ephraim none Korach, Shlomo Moran, Schmuel Zaks (1984)
```

Comment:
It's rather not "Ephraim none Korach", please correct

Reply:
Done